### ISSUE TITLE
Train the model

### ISSUE THREAD

**Issue by Zari222** [cofe-ai/fast-gector#16]
@Jason3900 
Hi, first thankyou for removing allennlp dependency from GECToR because i had problem for using that.
Actually I started to run this your version but had some problem and questions if you answer me I would be thankful.
1) I didn't used CUDA  and NVIDIA. Does it make problem in running. I should say I'm running the code in colab and using T4 gpu.
2) also I decided to just use part of the synthetic data to just see if it works but it is still about 2 hours and still running. Is it normal?
3) After it finished it gave this error:
![image](https://github.com/cofe-ai/fast-gector/assets/135531968/97200f12-869d-4644-a1ea-dd4c4cec29ec)
Some weights of RobertaModel were not initialized from the model checkpoint at roberta-base and are newly initialized: ['roberta.pooler.dense.weight', 'roberta.pooler.dense.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
2330722it [1:57:15, 26.72s/it][2023-10-25 16:50:16,631] [INFO] [launch.py:315:sigkill_handler] Killing subprocess 6518
[2023-10-25 16:50:16,632] [ERROR] [launch.py:321:sigkill_handler] ['/usr/bin/python3', '-u', 'train.py', '--local_rank=0', '--deepspeed', '--deepspeed_config', 'configs/ds_config_zero1_fp16.json', '--num_epochs', '10', '--max_num_tokens', '128', '--valid_batch_size', '256', '--cold_step_count', '0', '--warmup', '0.1', '--cold_lr', '1e-3', '--skip_correct', '0', '--skip_complex', '0', '--sub_token_mode', 'average', '--special_tokens_fix', '1', '--unk2keep', '0', '--tp_prob', '1', '--tn_prob', '1', '--detect_vocab_path', './data/vocabulary/d_tags.txt', '--correct_vocab_path', './data/vocabulary/labels.txt', '--do_eval', '--train_path', '/content/drive/MyDrive/GEC_test/gector/OUTPUT_FILE', '--valid_path', '/content/drive/MyDrive/GEC_test/gector/OUTPUT_Test', '--save_dir', 'ckpts/ckpt_20231025_14:52:16', '--use_cache', '0', '--log_interval', '1', '--eval_interval', '50', '--save_interval', '50', '--pretrained_transformer_path', 'roberta-base', '--tensorboard_dir', 'logs/tb/gector_20231025_14:52:16'] exits with return code = -9

4) Should I change "ckpt_path="ckpts/globalstep-xxxx"" in predict.sh ? What should it be?

Maybe my questions are too nave but I'm kind of new in GEC. So if you help me it will really help.

**Comment by Jason3900** [comment]
If it doesn't give more clues, you should probably check if there's any OOM errors in your system's log.

**Comment by Zari222** [comment]
okay I will check this.
Actually I'm working on GEC for another language than english. Do you think fast gector can be adopted to other language and how much data it need? because i want to create the dataset myself and maybe it wouldn't have much data.

**Comment by Jason3900** [comment]
It mainly depends on what downstream data you are facing. In a common sense, a few thousand of samples may be enough. But if your application varies a lot in the real scenario, it may requires more to achieve a better performance with robustness. Meanwhile, you can try to use wiki history to generate synthetic data.

**Comment by ZTurboX** [comment]
Have you experienced cpu memory increase until it explodes during training

**Comment by Jason3900** [comment]
I haven't encounter any CPU memory leaking issue for now, does it keep increase during training or in the process of loading the task dataset?

**Comment by ZTurboX** [comment]
> I haven't encounter any CPU memory leaking issue for now, does it keep increase during training or in the process of loading the task dataset?

during training

**Comment by ZTurboX** [comment]
> I haven't encounter any CPU memory leaking issue for now, does it keep increase during training or in the process of loading the task dataset?

small data may not be obvious to observe, large data have significantly observed cpu memory increase

**Comment by ZTurboX** [comment]
> I haven't encounter any CPU memory leaking issue for now, does it keep increase during training or in the process of loading the task dataset?

small data may not be obvious to observe, large data have significantly observed cpu memory increase

**Comment by Jason3900** [comment]
Does it occur in the training process or data loading process, and what is the size of the data you're using?

**Comment by ZTurboX** [comment]
occur in training process. 
two million data may  cpu memory increase but not  explode. 
five million data may  cpu memory increase until it explodes. 
Total cpu memory is 128g.

**Comment by Zari222** [comment]
> It mainly depends on what downstream data you are facing. In a common sense, a few thousand of samples may be enough. But if your application varies a lot in the real scenario, it may requires more to achieve a better performance with robustness. Meanwhile, you can try to use wiki history to generate synthetic data.

Thanks

**Comment by ZTurboX** [comment]
> occur in training process.
> two million data may cpu memory increase but not explode.
> five million data may cpu memory increase until it explodes.
> Total cpu memory is 128g.

occur in training process.
two million data may cpu memory increase but not explode.
five million data may cpu memory increase until it explodes.
Total cpu memory is 128g.
